src/services/users.services.js

```diff
@@ -0,0 +1,51 @@
+// const { ApiError } = require('../exceptions/ApiError');
+const { User } = require('../models/user');
+const { hashPassword } = require('../utils/password/hashPassword');
+
+const getUsers = async () => {
+  const users = await User.findAll();
+
+  return users;
+};
+
+const getUserByEmail = async (email) => {
+  const user = await User.findOne({ where: { email } });
+
+  return user;
+};
+
+const getUsersById = async (id) => {
+  const user = await User.findOne({ where: { id } });
+
+  return user;
+};
+
+const removeUser = async (id) => {
+  await User.destroy({ where: { id } });
+};
+
+const updateUser = async (user, userData) => {
+  Object.entries(userData).forEach(async ([field, data]) => {
+    if (data && field === 'password') {
+      user[field] = await hashPassword(data);
+    } else if (data) {
+      user[field] = data;
+    }
+  });
```

Comment:
Using `async` within `forEach` is not recommended because `forEach` does not handle asynchronous operations properly. Consider using a `for...of` loop or `Promise.all` to ensure that all asynchronous operations are completed before proceeding.